src/neo/SmartContract/StorageKey.cs

```diff
@@ -72,13 +84,34 @@ public override bool Equals(object obj)
 
         public override int GetHashCode()
         {
-            return Id.GetHashCode() + (int)Key.Murmur32(0);
+            return Id.GetHashCode() + (int)Key.Span.Murmur32(0);
         }
 
         void ISerializable.Serialize(BinaryWriter writer)
         {
-            writer.Write(Id);
-            writer.Write(Key);
+            if (cache != null)
+            {
+                writer.Write(cache);
+            }
+            else
+            {
+                writer.Write(Id);
+                writer.Write(Key.Span);
+            }
+        }
+
+        public byte[] ToArray()
+        {
+            if (cache is null)
+            {
+                using MemoryStream ms = new(sizeof(int) + Key.Length);
+                using BinaryWriter writer = new(ms, Utility.StrictUTF8, true);
+                writer.Write(Id);
+                writer.Write(Key.Span);
+                writer.Flush();
+                cache = ms.ToArray();
```

Comment:
For only two fields, it could be faster avoid the stream

```suggestion
                cache = new byte[sizeof(int) + Key.Length];
                BinaryPrimitives.WriteInt32LittleEndian(cache, Id);
                if (!Key.IsEmpty) Key.Span.CopyTo(cache.AsSpan(sizeof(int)));
```

Reply:
Updated.